enshrine-esdt-safe/src/lib.rs

```diff
@@ -24,9 +24,35 @@ pub trait EnshrineEsdtSafe:
     + common::storage::CommonStorage
 {
     #[init]
-    fn init(&self, is_sovereign_chain: bool) {
+    fn init(
+        &self,
+        is_sovereign_chain: bool,
+        wegld_identifier: Option<ManagedBuffer>,
+        sov_token_prefix: Option<ManagedBuffer>,
+    ) {
         self.is_sovereign_chain().set(is_sovereign_chain);
         self.set_paused(true);
+
+        if !is_sovereign_chain {
+            match (wegld_identifier, sov_token_prefix) {
```

Comment:
This looks very hard to maintain if you have a lot of arguments. 
- it would be better to use `OptionalValue` instead of `Option`
- just take the arguments one by one and `sc_panic!` if error
- don't forget to throw error if wegld identifier is invalid

Reply:
you can't use 2 OptionalValues as parameters